packages/yarnpkg-libzip/sources/sync.ts

```diff
@@ -1,17 +1,20 @@
-import createModule            from './libzipSync';
-import {Libzip, makeInterface} from './makeInterface';
+import {setFactory, getInstance} from './instance';
+import createModule              from './libzipSync';
+import {makeInterface}           from './makeInterface';
 
-let mod: Libzip | null = null;
+export * from './common';
 
-export function getLibzipSync() {
-  if (mod === null)
-    mod = makeInterface(createModule());
+setFactory(() => {
```

Comment:
Can side effects be avoided here?  This will remove the safety to tree-shake this module.

Reply:
I don't see a way to avoid it; however I can mark this file as having side effects, so that other files will remain tree-shaken

Reply:
Actually the package doesn't list `sideEffects` at all at the moment, so I'm not sure it's tree-shakable 🤔 I'm not too sure what should be the proper value (do we even have to list the entry point as having side effects, since it'll always be included if anything else is included?), so I won't touch it in this diff